### ISSUE TITLE
Account name can block summon alts of same name.

### ISSUE THREAD

**Issue by JakalairVG** [liyunfan1223/mod-playerbots#643]
**Describe the bug**
If the an account name and player name are the same it can cause problems summoning the bot as an alt.

**Commit hash**
2e37c1ff1d7fb36d5a003edc44a15a939a2d4667

**To Reproduce**
1. create 2 accounts one named "george" and the other "tarzan", neither of them GM accounts. (any names work)
2. on account "george" create two characters named "jane" and "tarzan"
3. login on jane and attempt to use the command ".playerbots bot add tarzan" and it will not work.
4. log out of jane and into tarzan and try to summon jane ".playerbots bot add jane" and it will work.

**Expected behavior**
It should check and see the account that is not accessible and then check the player after.
Thus even if there is an account and player with the same name it will not conflict with each other.
Either that or a way to specify that I want to add an account or a player.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Chrome
 - 130.0.6723.70

**Additional context**
I stumbled across this while setting up the server and for some reason I could never summon in one character. Turns out I had named the GM account the same name as that character not thinking about it. I did some more testing just to see if it was unbot, but even the console command was not doing anything.

**Comment by Bobblybook** [comment]
Confirmed, I had a similar issue - character name the same as acct name and I had no way to reference the character only, it would invite the whole acct at once and form a raid.

I had no solution currently and had to rename my account to something else, but something does need to implemented to be able to explicitly specify acct or char.

**Comment by Dreathean** [comment]
Yeah I think it would make sense to remove accounts from ".playerbot bot add <name>" and make a separate ".playerbot bot add acct <name>" for adding a whole account at once.

**Comment by xSparky911x** [comment]
I am currently working on a fix for this changing the command to .playerbots bot add <account/character> <name> as well as same for remove command. what do you think about that?

**Comment by Dreathean** [comment]
I would personally prefer to keep them separated, primarily because some people (like myself) want to add an entire account at once. For example, being able to login my entire raid team of 24 bots with one macro using 3 lines for 3 accounts.

If you do it like that, that won't be possible anymore as you'd need to include every individual bot every time.

**Comment by xSparky911x** [comment]
> I would personally prefer to keep them separated, primarily because some people (like myself) want to add an entire account at once. For example, being able to login my entire raid team of 24 bots with one macro using 3 lines for 3 accounts.
> 
> If you do it like that, that won't be possible anymore as you'd need to include every individual bot every time.

Well no you wouldn't. If you want to add an entire account you simply use
```.playerbots bot add account <account name>```
Want to add just a toon then you use
```.playerbots bot add character <character name>```

With the current system there is a bug. If the account name is JohnDoe and that account has 25 toons, one of which is named JohnDoe as well. When a player tries to add just the toon "JohnDoe" and issues the command ```.playerbots bot add JohnDoe```. They get JohnDoe and 24 other toons even though they really only wanted just 1 toon named JohnDoe because the current system looks for the account name first and if it finds one adds all those toons associated with the account. 

With the proposed "fix" if the player only wants the toon "JohnDoe" they simply use
```.playerbots bot add character JohnDoe```
If they want all of JohnDoes' 25 toons they would use 
```.playerbots bot add account JohnDoe```

**Comment by Dreathean** [comment]
I apologize, I misread too quickly and thought you would have to specify both account name and character name for each one. Sounds like a great addition, thank you.

**Comment by xSparky911x** [comment]
Haha yea I figured after I replied. But yes they will be seperate

**Comment by xSparky911x** [comment]
Ok I think I have a working version of this functionality now however it will break addons like Multibot and Unbot due to the command change and they will have to be updated as well

**Comment by xSparky911x** [comment]
> > Ok I think I have a working version of this functionality now however it will break addons like Multibot and Unbot due to the command change and they will have to be updated as well
> 
> Can you create the PR? If other mods need changes, so be it.

Sorry I was stuck on fixing other issues and was away from this for some time now but I am ready to get working at it again. I will keep you updated and let you know

**Comment by xSparky911x** [comment]
This can be marked as closed or fixe now in #1070

**Comment by Dreathean** [comment]
Got it - thank you Sparky!
